### PR TITLE
Simulation for ROS 2 Client

### PR DESCRIPTION
**Pull request by aaronchongth**
### Implemented feature

This PR includes a simulation example for the newly merged ROS 2 Client, and documentation on using it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,95 +130,145 @@ colcon build --cmake-args -DCMAKE_BUILD_TYPE=RELEASE
...
 roslaunch ff_examples_ros1 fake_client.launch
 ```
 
-The client will then start subscribing to all the necessary topics, and start publishing robot states over DDS to the server. Start the server using
+This example emulates a running ROS 2 robot,
 
 ```bash
-source ~/server_ws/install/setup.bash
+source ~/ff_ros2_ws/install/setup.bash
+ros2 launch ff_examples_ros2 fake_client.launch
```
Comment (cnboonhan) [inline README.md:144]: would this have a .xml postfix?

**Review by cnboonhan — CHANGES_REQUESTED** [review]
this is amazing! brought it on a test run and everything looks great. tested pauses and resumes too. 

I cant comment on the ros2 navigation but i think its all right. Also not sure how to test docking?

File: README.md (hunk 1 above)
Comment (aaronchongth) [reply README.md]: Thanks for catching that, fixed in https://github.com/open-rmf/free_fleet/pull/107/commits/ce9169088436cf5fc64d558d8dafb74804d1ffe4

**Review by cnboonhan — APPROVED** [review]
looks amazing! Looking forward to building on these in the future 👍

**Comment by vkuehn** [comment]
does this now not work anymore with foxy ?

**Comment by cnboonhan** [comment]
hello,  the main branch of free fleet would not build with foxy anymore, as there were breaking api changes during the shift to galactic

**Comment by aaronchongth** [comment]
Hello @vkuehn! Currently only the `free_fleet_client_ros2`  has breaking changes that require ROS 2 Galactic. One way to keep compiling `main` with ROS 2 Foxy would be skipping that package, 
```
colcon build --packages-skip free_fleet_client_ros2
```

**Pull request merged**